misk/src/main/kotlin/misk/backoff/Retries.kt

```diff
@@ -0,0 +1,34 @@
+package misk.backoff
+
+/**
+ * Retries the provided function up to a certain number of times, applying the given backoff
+ * between each retry. The retry function is provided with current retry count, in case this is
+ * relevant
+ */
+fun <A> retry(upTo: Int, withBackoff: Backoff, f: (Int) -> A): A {
+  require(upTo > 0) { "must support at least one call" }
+
+  withBackoff.reset()
+
+  var lastException: Exception? = null
+
+  for (i in 0 until upTo) {
+    try {
+      val result = f(i)
+      withBackoff.reset()
+      return result
+    } catch (e: Exception) {
+      lastException = e
+
+      if (i + 1 < upTo) {
+        try {
+          Thread.sleep(withBackoff.nextRetry().toMillis())
+        } catch (e: InterruptedException) {
+          throw e
```

Comment:
why not continue? I don't have a lot of knowledge of why this would happen

Reply:
``InterruptedException`` is thrown when someone explicitly asks that the thread interrupt its work and not continue (e.g. shutdown now plz).  It's a signal that the thread should stop whatever it is doing.